File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/mappers/BulletsToBulletStatesMapper.java
package dev.robocode.tankroyale.server.mappers;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import dev.robocode.tankroyale.server.model.Bullet;
import dev.robocode.tankroyale.schema.BulletState;

@SuppressWarnings("WeakerAccess")
public final class BulletsToBulletStatesMapper {

	private BulletsToBulletStatesMapper() {}

	public static List<BulletState> map(Set<Bullet> bullets) {
		List<BulletState> bulletStates = new ArrayList<>();
		for (Bullet bullet : bullets) {
			bulletStates.add(BulletToBulletStateMapper.map(bullet));
		}
		return bulletStates;
	}
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/mapper/BulletsToBulletStatesMapper.kt
package dev.robocode.tankroyale.server.mapper

import dev.robocode.tankroyale.schema.game.BulletState
import dev.robocode.tankroyale.server.model.IBullet

object BulletsToBulletStatesMapper {
    fun map(bullets: Set<IBullet>): List<BulletState> {
        val bulletStates = mutableListOf<BulletState>()
        bullets.forEach { bulletStates += BulletToBulletStateMapper.map(it) }
        return bulletStates
    }
}